ecosystem/typescript/sdk/README.md

```diff
@@ -4,130 +4,78 @@
+- You can view the generated [Type Doc](https://aptos-labs.github.io/ts-sdk-doc/) for the latest release of the SDK.
+- For in-depth examples, check out the [examples](./examples) folder with ready-made `package.json` files to get you going quickly!
 
-For Javascript or Typescript usage, check out the [`./examples`][examples] folder with ready-made `package.json` files to get you going quickly!
+### Development environment setup
 
-If you are using the types in a `commonjs` module, like in a Node app, you just have to enable `esModuleInterop`
-and `allowSyntheticDefaultImports` in your `tsconfig` for types compatibility:
+Setup an `.env` file to configure the URLs.
+From the [root](https://github.com/aptos-labs/aptos-core/tree/main/ecosystem/typescript/sdk) of this package, run:
 
-```json
-{
-  ...
-  "compilerOptions": {
-    "allowSyntheticDefaultImports": true,
-    "esModuleInterop": true
-    ...
-  }
-}
+```ts
+rm .env
+echo 'APTOS_NODE_URL="http://localhost:8080/v1"' >> .env
+echo 'APTOS_FAUCET_URL="http://localhost:8081"' >> .env
 ```
 
-### Requirements
+### Testing
 
-- [Node.js](https://nodejs.org)
-- [Yarn](https://pnpmpkg.com/)
+To run the full SDK tests, From the [root](https://github.com/aptos-labs/aptos-core/tree/main/ecosystem/typescript/sdk) of this package, run:
 
-```bash
-pnpm install
+```ts
+pnpm test
 ```
 
-### Generating API client
-
-This SDK is composed of two parts, a core client generated from the OpenAPI spec of the API, and a set of wrappers that make it nicer to use, enable certain content types, etc.
+> If you see strange behavior regarding HTTP clients, try running the tests with `--detectOpenHandles`.
 
-To generate the core client from the spec, run:
+To test a single file in the SDK, From the [root](https://github.com/aptos-labs/aptos-core/tree/main/ecosystem/typescript/sdk) of this package, run:
 
-```bash
-pnpm generate-client
+```ts
+npx jest -- <path/to/file.test.ts>
 ```
 
-### Working with devnet
+To use the local build in a local project:
 
-See the quickstart above.
+```ts
+// run from the root of this package
+pnpm build
+// run on your local project
+pnpm add PATH_TO_LOCAL_SDK_PACKAGE
+```
 
 ### Working with local node
 
 To develop in a local environment, you need to use the SDK from the [main](https://github.com/aptos-labs/aptos-core/tree/main/ecosystem/typescript/sdk) branch.
 
-**NOTE**
-SDK from the main branch might not be compatible with the devnet.
-
-Run a local node (run from the root of the repo):
+Run a local node (run from the root of the [repo](https://github.com/aptos-labs/aptos-core/)):
 
-```
+```ts
 cargo run -p aptos -- node run-local-testnet --with-faucet --faucet-port 8081 --force-restart --assume-yes
 ```
 
-Run the SDK tests and make sure they pass. Go to the SDK directory, and setup an env to configure the URLs:
-
-```
-rm .env
-echo 'APTOS_NODE_URL="http://127.0.0.1:8080/v1"' >> .env
-echo 'APTOS_FAUCET_URL="http://127.0.0.1:8081"' >> .env
```

Comment:
aren't these still kinda needed how does one connect to their local node?

Reply:
yes! and we still have it here https://github.com/aptos-labs/aptos-core/pull/8362/files#diff-5d279b45707e2394d3480bbc09738d4354c0bd20404b6ed78d01c4d7106c58e0R38